adapter/utils/src/test/java/org/mobilitydata/gtfsvalidator/timeutils/TimeUtilsImplTest.java

```diff
@@ -240,17 +240,68 @@ void convertHHMMSStoIntegerAndIntegerToHHMMSSShouldBeConsistent() {
     }
 
     @Test
-    void overlappingPeriodsShouldReturnTrue() {
+    void partialOverlapWithOneCommonBoundaryShouldReturnTrue() {
+        //   a |--------|         07:00 am -> 10:00 am
+        //   b     |----|         09:00 am -> 10:00 am
+        // period `b` partially overlaps with period `a`
+        assertTrue(TIME_CONVERSION_UTILS.arePeriodsOverlapping(700, 1000,
+                900, 1000));
+
+        // permute periods `a` and `b` in method call
+        assertTrue(TIME_CONVERSION_UTILS.arePeriodsOverlapping(900, 1000,
+                700, 1000));
+
+        //   a |--------|         07:00 am -> 10:00 am
+        //   b |----|             07:00 am -> 09:00 am
+        // period `b` partially overlaps with period `a`
+        assertTrue(TIME_CONVERSION_UTILS.arePeriodsOverlapping(700, 1000,
+                700, 900));
+
+        // permute periods `a` and `b` in method call
+        assertTrue(TIME_CONVERSION_UTILS.arePeriodsOverlapping(700, 900,
+                700, 1000));
+    }
+
+    @Test
+    void partialOverlapPeriodsShouldReturnTrue() {
```

Comment:
I think you're missing the opposite of this test?

```
b |--------| 
      a  |----|
```

Reply:
Oh, nevermind, I see it below. Could you add another comment visual above that test to explain?

Reply:
Same for the other methods where you swap numbers for the permutation at the end.